### PR TITLE
Support invoking the Fast Dependency Scanner: `-scan-dependencies`

### PR DESCRIPTION
**Pull request by artemcm**
Add support for the additional compiler mode added in: #28515.

### CODE REVIEW

File: Sources/SwiftDriver/Utilities/FileType.swift [hunk 1]
```diff
@@ -87,6 +87,9 @@ public enum FileType: String, Hashable, CaseIterable, Codable {
   /// Text-based dylib (TBD) file.
   case tbd
 
+  /// JSON-based Module Dependency Scanner output
+  case JSONDependencies = "dependencies.json"
```
Comment (DougGregor) [inline Sources/SwiftDriver/Utilities/FileType.swift:91]: I hate to make this one of my few comments, but can you call this `jsonDependencies`? Values in Swift start with lowercase, including initialisms like this.

File: Tests/SwiftDriverTests/SwiftDriverTests.swift [hunk 2]
```diff
@@ -1586,6 +1586,16 @@ final class SwiftDriverTests: XCTestCase {
     }
   }
 
+  func testScanDependenciesOption() throws {
+    do {
+      var driver = try Driver(args: ["swiftc", "-scan-dependencies", "foo.swift"])
```
Comment (DougGregor) [inline Tests/SwiftDriverTests/SwiftDriverTests.swift:1591]: `-scan-dependencies` supports other kinds of outputs. Can you feed in arguments to ask it to produce a dependencies file (`.d`) and make sure that ends up on the resulting command line?

**Comment by DougGregor** [comment]
@swift-ci please test

**Comment by DougGregor** [comment]
@swift-ci please test

**Comment by DougGregor** [comment]
@swift-ci test

**Comment by artemcm** [comment]
Just needed a rebase.
@swift-ci test

**Comment by DougGregor** [comment]
@swift-ci test

**Pull request merged**